deployment/configenv/xml_jlibpt/Software.cpp

```diff
@@ -0,0 +1,295 @@
+   return getSWComp(comp)->remove(params);
+}
+
+void Software::getSWCompName(const char *inputName, StringBuffer& out)
+{
+   const char * compNameLC = (StringBuffer(inputName).toLowerCase()).str();
+
+   // Return xsd name in buildset.xml
+   out.clear();
+   if (!stricmp(compNameLC, "directories")  || !stricmp(compNameLC, "dirs"))
+   {
+      out.append("directories");
+   }
+   else if (!stricmp(compNameLC, "dali") || !stricmp(compNameLC, "DaliServerProcess"))
+   {
+      out.append("dali");
+   }
+   else if (!stricmp(compNameLC, "dropzone"))
+   {
+      out.append("DropZone");
+   }
+   else if (!stricmp(compNameLC, "roxie") || !stricmp(compNameLC, "RoxieCluster"))
+   {
+      out.append("roxie");
+   }
+   else if (!stricmp(compNameLC, "thor") || !stricmp(compNameLC, "ThorCluster"))
+   {
+      out.append("thor");
+   }
+   else if (!stricmp(compNameLC, "esp") || !stricmp(compNameLC, "EspProcess"))
+   {
+      out.append("esp");
+   }
+   else if (!stricmp(compNameLC, "elcwatch") || !stricmp(compNameLC, "espsmc"))
+   {
+      out.append("espsmc");
+   }
+   else if (!stricmp(compNameLC, "esdl") || !stricmp(compNameLC, "DynamicESDL"))
+   {
+      out.append("DynamicESDL");
+   }
+   else if (!stricmp(compNameLC, "ws_sql") || !stricmp(compNameLC, "wssql"))
+   {
+      out.append("ws_sql");
+   }
+   else if (!stricmp(compNameLC, "ws_ecl") || !stricmp(compNameLC, "wsecl"))
+   {
+      out.append("ws_ecl");
+   }
+   else if (!stricmp(compNameLC, "wslogging"))
+   {
+      out.append("wslogging");
+   }
+   else if (!stricmp(compNameLC, "cassandra") || !stricmp(compNameLC, "CassandraLoggingAgent"))
+   {
+      out.append("cassandraloggingagent");
+   }
+   else if (!stricmp(compNameLC, "esplogging") || !stricmp(compNameLC, "EspLoggingAgent"))
+   {
+      out.append("esploggingagent");
+   }
+   else if (!stricmp(compNameLC, "loggingmgr") || !stricmp(compNameLC, "loggingmanager"))
+   {
+      out.append("loggingmanager");
+   }
+   else if (!stricmp(compNameLC, "agent") || !stricmp(compNameLC, "eclagent") ||
+         !stricmp(compNameLC, "EclAgentiProcess"))
+   {
+      out.append("eclagent");
+   }
+   else if (!stricmp(compNameLC, "eclccsrv") || !stricmp(compNameLC, "eclccserver") ||
+            !stricmp(compNameLC, "eclcc") || !stricmp(compNameLC, "EclCCServerProcess"))
+   {
+      out.append("eclccserver");
+   }
+   else if (!stricmp(compNameLC, "sch") || !stricmp(compNameLC, "eclsch") ||
+            !stricmp(compNameLC, "scheduler") || !stricmp(compNameLC, "EclCCSchedulerProcess"))
+   {
+      out.append("eclscheduler");
+   }
+   else if (!stricmp(compNameLC, "dfu") || !stricmp(compNameLC, "dfusrv") ||
+            !stricmp(compNameLC, "dfuserver") || !stricmp(compNameLC, "DfuServerProcess"))
+   {
+      out.append("dfuserver");
+   }
+   else if (!stricmp(compNameLC, "topo") || !stricmp(compNameLC, "Topology"))
+   {
+      out.append("topology");
+   }
+   else
+   {
+      out.append(compNameLC);
+   }
```

Comment:
This is another candidate for a map, as is the next method.

Reply:
This method map the various user input name to internal component name. The next method use this to resolve the component name and return component object.